tests/unit/time_series_test.coffee

```diff
@@ -71,3 +72,176 @@ test 'Margins are the right size when showLegend is no', (assert) ->
   assert.equal @subject().get('marginRight'), 0, 'no right margin'
   assert.equal @subject().get('marginBottom'), 0,
     'no bottom margin if showLegend is no'
+
+timeSeriesContentLineNonDynamic =
+  yAxisFromZero: no
+  dynamicXAxis: no
+  selectedInterval: ''
+  minTimeSpecificity: 'S'
+  maxTimeSpecificity: 'Y'
+  lineData: []
+
+test 'The number of labels should be equal to the number of day intervals',
+(assert) ->
+  content = $.extend({}, timeSeriesContentLineNonDynamic)
+  content.lineData = [
+      time: d3.time.format('%Y-%m-%d').parse "2011-04-1"
+      value: 1
+    ,
+      time: d3.time.format('%Y-%m-%d').parse "2011-04-10"
+      value: 2
+  ]
+  content.selectedInterval = 'D'
+  # The amount the number of labels can be off by to still be correct
+  BUFFER = 3
+  @subject content
+
+  maxLabels = @subject().get('maxNumberOfLabels')
+  if maxLabels < 9
+    # this case occurs if the graphical window is too narrow and
+    # the number of ticks defaults to the number of x ticks
+    assert.equal @subject().get('labelledTicks').length <= maxLabels,
+      true, 'The number of labelled days is less than or equal to
+      the max number of labels'
+  else
+    assert.equal Math.abs(@subject().get('labelledTicks').length - 9) <= BUFFER,
+      true, 'The number of labelled days is correct'
+
+test 'The number of labels should be equal to the number of month intervals', 
+(assert) ->
+  content = $.extend({}, timeSeriesContentLineNonDynamic)
+  content.lineData = [
+      time: d3.time.format('%Y-%m-%d').parse "2011-04-1"
+      value: 1
+    ,
+      time: d3.time.format('%Y-%m-%d').parse "2011-12-07"
+      value: 2
+  ]
+  content.selectedInterval = 'M'
+  # The amount the number of labels can be off by to still be correct
+  BUFFER = 3
+  @subject content
+
+  maxLabels = @subject().get('maxNumberOfLabels')
+  if maxLabels < 9
+    # this case occurs if the graphical window is too narrow an
+    # the number of ticks defaults to the number of x ticks
+    assert.equal @subject().get('labelledTicks').length <= maxLabels,
+      true, 'The number of labelled months is less than or equal to
+      the max number of labels'
+  else
+    assert.equal Math.abs(@subject().get('labelledTicks').length - 9) <= BUFFER,
+      true, 'The number of labelled months is correct'
+
+
+# Put a small time interval like days with a ridiculously long time span like
+# 30 years and make sure the number of labels doesn't exceed the max number
+# of labels
+test 'The number of labels should not exceed the max number of labels',
+(assert) ->
+  content = $.extend({}, timeSeriesContentLineNonDynamic)
+  content.lineData = [
+      time: d3.time.format('%Y-%m-%d').parse "2011-04-1"
+      value: 1
+    ,
+      time: d3.time.format('%Y-%m-%d').parse "2040-12-07"
+      value: 2
```

Comment:
this is copy pasted in this test and the two tests before it, could you make a variable for it and just use that variable?